frame/contracts/src/exec.rs

```diff
@@ -1328,14 +1329,19 @@ where
 		&mut self.top_frame_mut().nested_gas
 	}
 
-	fn append_debug_buffer(&mut self, msg: &str) -> bool {
+	fn append_debug_buffer(&mut self, msg: &str) -> Result<bool, ()> {
```

Comment:
Return the appropriate `DispatchError` here instead of `()`. Should be a new error instead of `OutOfBounds`. This functions knows the error reason. Not the caller.